crowd_anki/github/github_importer.py

```diff
@@ -34,21 +35,35 @@ def import_from_github(self):
         if repo and ok:
             self.download_and_import(repo)
 
-    def download_and_import(self, repo):
+    def download_and_import(self, github_repo):
+        deck_base_name = github_repo.split("/")[-1]
         try:
-            response = urllib2.urlopen(GITHUB_LINK.format(repo))
-            response_sio = StringIO.StringIO(response.read())
-            with zipfile.ZipFile(response_sio) as repo_zip:
-                repo_zip.extractall(tempfile.tempdir)
-
-            deck_base_name = repo.split("/")[-1]
-            deck_directory_wb = Path(tempfile.tempdir).joinpath(deck_base_name + "-" + BRANCH_NAME)
-            deck_directory = Path(tempfile.tempdir).joinpath(deck_base_name)
-            utils.fs_remove(deck_directory)
-            deck_directory_wb.rename(deck_directory)
-            # Todo progressbar on download
-
-            AnkiJsonImporter.import_deck(self.collection, deck_directory)
+            try:
+                repo = None
+                try:
+                    repo = Repo(Path(self.collection.media.dir()).joinpath("..", "CrowdAnkiGit", deck_base_name, "git"))
+                    repo.remote("origin").pull()
+                except git.exc.InvalidGitRepositoryError: # New repository
+                    repo = Repo.clone_from(GITHUB_REPO.format(github_repo), Path(self.collection.media.dir()).joinpath("..", "CrowdAnkiGit", deck_base_name, "git"))
+                AnkiJsonImporter.import_deck(self.collection, Path(repo.working_tree_dir))
+            
+            except git.exc.GitCommandNotFound: # Git not avaiable. Use .zip archive instead
+                QErrorMessage.qtHandler().showMessage('Error accessing the git executable.\n'
```

Comment:
At this point, as you don't provide any new functionality besides zip version and don't insure presence of git exec - I would think it would be more reasonable to silently (without prompting user, but with writing to log) fall back to zip here.